Login

iplonk3 opened the pull request:
Authenticates through edge service instead of directly going to authentication service and gets user id from access token for SSE

Commit:
Initial work on login

Commit:
Use user id

Commit:
Get userid from access token

Commit:
Fixed merge conflicts

Commit:
Fix merge conflicts in trip planner file

dmssargent commented on docker-compose.yml:
```diff
@@ -0,0 +1,82 @@
+      MYSQL_DATABASE: 'user_service'
+
+  userservice:
+    build: ~/Documents/ai-testing-org/authentication-service/
```
This should pull from an image

dmssargent commented on docker-compose.yml:
```diff
@@ -0,0 +1,82 @@
+    depends_on:
+      - discoveryservice
+      - rabbitmq
+    build: ~/go/src/github.com/AITestingOrg/calculation-service/
```
This should pull from an image

dmssargent commented on docker-compose.yml:
```diff
@@ -0,0 +1,82 @@
+    depends_on:
+      - discoveryservice
+      - rabbitmq
+    build: ~/Documents/ai-testing-org/gmaps-adapter/
```
This should pull from an image

dmssargent commented on src/app/common/states/authentication.service.ts:
```diff
@@ -0,0 +1,69 @@
+@Injectable()
+export class AuthenticationService {
+
+    edgeServiceUrl = `http://localhost:8080/api/userservice/auth/oauth/token`;
```
The field name reflects just the edge service, however the full path routes to the user service. Can you update the field name to denote this?

iplonk3 replied:
Yes good suggestion

dmssargent commented on src/app/common/states/authentication.service.ts:
```diff
@@ -0,0 +1,69 @@
+            },
+            err => {
+                alert('Invalid credientals');
+                console.warn(`Failed to communicated with the user service. Err: ${JSON.stringify(err)}`);
```
Those are two unrelated error messages, how can you know if the credentials are invalid if you failed to communicate with the user service?

dmssargent commented on src/app/common/states/authentication.service.ts:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    logout() {
+        localStorage.clear();
```
You are only using the 'accessToken' item, but you are wiping everything from local storage, can logout just delete what it needs to?

dmssargent commented on src/app/passengers/components/smart/trip-planner/trip-planner.component.ts:
```diff
@@ -304,12 +278,11 @@ export class TripPlannerComponent implements OnInit {
       'origin': this.pickupInputElementRef.nativeElement.value,
       'destination': this.pickupOutputElementRef.nativeElement.value,
-      'userId': '560c62f4-8612-11e8-adc0-fa7ae01bbebc'
+      'userId': localStorage.getItem('userId')
```
Can we just ask the authentication service for this value via a new function, so we aren't leaking the implementation details?

iplonk3 replied:
Yes good suggestion

Commit:
Initial work on unit tests for authentication service

Commit:
Add tests for routing

Commit:
Update mock router

Commit:
Fix mock HTTP requests

Commit:
Mock routing properly

Commit:
Remove docker-compose file 

This file was created for testing purposes only and should not be included in this project.

jusoto committed:
Update Google API Key

jusoto committed:
Merge branch 'login' of https://github.com/AITestingOrg/front-end into login

jusoto committed:
Merge branch 'master' of https://github.com/AITestingOrg/front-end into login

# Conflicts:
#	package-lock.json
#	package.json
#	src/app/app.module.ts
#	src/app/common/states/notification.service.ts
#	src/app/passengers/components/smart/trip-planner/trip-planner.component.ts
#	yarn.lock

jusoto committed:
Resolve merging conflicts

jusoto committed:
Added docker-compose-local.yml

jusoto committed:
Added docker to .travis.yml

jusoto committed:
Replaced angular2-jwt dependency by @auth0/angular-jwt due to incompatibility with angular 6

jusoto committed:
Ammend to docker-compose-local.yml: authentication-service replaced by user-service

jusoto committed:
Modified travis.yml without docker-compose script
